src/com/sap/piper/Utils.groovy

```diff
@@ -72,67 +75,29 @@ def unstashAll(stashContent) {
+    }
+    return sb.toString()
 }
 
 void pushToSWA(Map parameters, Map config) {
     try {
-        //allow opt-out via configuration
-        if (!config?.collectTelemetryData) {
-            echo "[${parameters.get('step')}] Telemetry Report to SWA disabled!"
-            return
-        }
-
-        def swaCustom = [:]
-
-        /* SWA custom parameters:
-            custom3 = step name (passed as parameter step)
-            custom4 = job url hashed (calculated)
-            custom5 = build url hashed (calculated)
-            custom10 = stage name
-            custom11 = step related parameter 1 (passed as parameter stepParam1)
-            custom12 = step related parameter 2 (passed as parameter stepParam2)
-            custom13 = step related parameter 3 (passed as parameter stepParam3)
-            custom14 = step related parameter 4 (passed as parameter stepParam4)
-            custom15 = step related parameter 5 (passed as parameter stepParam5)
-        */
-
-        def swaUrl = 'https://webanalytics.cfapps.eu10.hana.ondemand.com/tracker/log'
-        def action_name = 'Piper Library OS'
-        def idsite = '827e8025-1e21-ae84-c3a3-3f62b70b0130'
-        def url = 'https://github.com/SAP/jenkins-library'
-        def event_type = parameters.get('eventType') ?: 'library-os'
-
-        swaCustom.custom3 = parameters.get('step')
-        swaCustom.custom4 = generateSha1Inline(env.JOB_URL)
-        swaCustom.custom5 = generateSha1Inline(env.BUILD_URL)
-        swaCustom.custom10 = parameters.get('stageName')
-        swaCustom.custom11 = parameters.get('stepParam1')
-        swaCustom.custom12 = parameters.get('stepParam2')
-        swaCustom.custom13 = parameters.get('stepParam3')
-        swaCustom.custom14 = parameters.get('stepParam4')
-        swaCustom.custom15 = parameters.get('stepParam5')
-
-        def options = []
-        options.push("-G")
-        options.push("-v \"${swaUrl}\"")
-        options.push("--data-urlencode \"action_name=${action_name}\"")
-        options.push("--data-urlencode \"idsite=${idsite}\"")
-        options.push("--data-urlencode \"url=${url}\"")
-        options.push("--data-urlencode \"event_type=${event_type}\"")
-        for(def key : ['custom3', 'custom4', 'custom5', 'custom10', 'custom11', 'custom12', 'custom13', 'custom14', 'custom15']){
-            if (swaCustom[key] != null) options.push("--data-urlencode \"${key}=${swaCustom[key]}\"")
-        }
-        options.push("--connect-timeout 5")
-        options.push("--max-time 20")
-
-        sh(returnStatus: true, script: "#!/bin/sh +x\ncurl ${options.join(' ')} > /dev/null 2>&1 || echo '[${parameters.get('step')}] Telemetry Report to SWA failed!'")
+        parameters.actionName = parameters.get('actionName') ?: 'Piper Library OS'
+        parameters.eventType = parameters.get('eventType') ?: 'library-os'
+        parameters.jobUrlSha1 =  generateSha1(env.JOB_URL)
+        parameters.buildUrlSha1 = generateSha1(env.BUILD_URL)
```

Comment:
I would vote for moving all the logic from here to the Telemetry class so that we can deprecated this method and use `Telemetry` directly.

Reply:
We keep this here to shield the `env.JOB_URL` from the extensions.